CONTRIBUTING.md

```diff
@@ -1,34 +1,70 @@
+1. [Fork the OpenDAL repository](https://github.com/apache/incubator-opendal/fork) in your own GitHub account.
+1. [Create a new Git branch](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-and-deleting-branches-within-your-repository).
+1. Make your changes.
+1. [Submit the branch as a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) to the main OpenDAL repo. An OpenDAL team member should comment and/or review your pull request within a few days. Although, depending on the circumstances, it may take longer.
+
+## Workflow
+
+### Git Branches
+
+*All* changes must be made in a branch and submitted as [pull requests](#github-pull-requests). OpenDAL does not adopt any type of branch naming style, but please use something descriptive of your changes.
 
-## Communication
+### GitHub Pull Requests
 
-We use [GitHub Discussions](https://github.com/apache/incubator-opendal/discussions) for general discussions and questions. Please use the `Q&A` category for questions and the `General` category for general discussions.
+Once your changes are ready you must submit your branch as a [pull request](https://github.com/apache/incubator-opendal/pulls).
 
-We also use [GitHub Issues](https://github.com/apache/incubator-opendal/issues) for bug reports and feature requests.
+#### Title
 
-Otherwise, you can join our [Discord Channel](https://discord.gg/XQy8yGR2dg) for real-time discussions.
+The pull request title must follow the format outlined in the [conventional commits spec](https://www.conventionalcommits.org). [Conventional commits](https://www.conventionalcommits.org) is a standardized format for commit messages. OpenDAL only requires this format for commits on the `main` branch. And because OpenDAL squashes commits before merging branches, this means that only the pull request title must conform to this format.
 
-## Taking Issues
+The following are all good examples of pull request titles:
 
-It is highly recommended to request an assignment before beginning work on any issues to avoid conflicts with other community members.
+```text
+feat(services/gcs): Add start-after support for list
+docs: add hdfs classpath related troubleshoot
+ci: Mark job as skipped if owner is not apache
+fix(services/s3): Ignore prefix if it's empty
+refactor: Polish the implementation of webhdfs
+```
 
-It is encouraged to post your work progress in the issues comment so that the community can stay informed about ongoing developments and offer any necessary assistance.
+#### Reviews & Approvals
 
-It is possible that your assignment has been removed after 14 days of no updates. Please note that this is not a punishment or dissatisfaction from the community. We understand that life can be challenging, and things may not always go as planned.
+All pull requests should be reviewed by at least one OpenDAL committers.
```

Comment:
```suggestion
All pull requests should be reviewed by at least one maintainer.
```

Reply:
No. Committers != maintainer.

Reply:
Not everyone understands the role of ASF. The `maintainer` is more accessible. 🤔